File: ClassesDemo/src/Period.java
import java.time.Duration;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public abstract class Period
{
    protected String description;
    protected LocalDateTime startTime;
    protected LocalDateTime endTime;
    protected double hours;

    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");

    public Period(String description)
    {
        this.description = description;
        this.startTime = LocalDateTime.now();
        this.endTime = LocalDateTime.now();
        this.hours = 0;
    }

    // ===========================================================================

    public void updateHours()
    {
        Duration duration = Duration.between(getStartTime(), getEndTime());

        double hours = duration.toHours();
        double minutes = duration.toMinutes();
        hours += (minutes % 60 / 60.0);

        this.hours = hours;
    }

    // ===========================================================================

    public String getDescription()
    {
        return description;
    }

    public LocalDateTime getStartTime()
    {
        return startTime;
    }

    public LocalDateTime getEndTime()
    {
        return endTime;
    }

    public double getHours()
    {
        return hours;
    }

    // ===========================================================================

    public void setDescription(String description)
    {
        this.description = description;
    }

    public void setStartTime(LocalDateTime startTime)
    {
        this.startTime = startTime;
    }

    public void setEndTime(LocalDateTime endTime)
    {
        this.endTime = endTime;
    }

    public void setHours(double hours)
    {
        this.hours = hours;
    }

    // ===========================================================================

    @Override
    public String toString()
    {
        return "Description: " + description + "\n"
        + "Start Time: " + startTime.format(FORMATTER) + "\n"
        + "End Time: " + endTime.format(FORMATTER) + "\n"
        + "Duration In Hours: " + String.format("%.2f", hours);
    }
}

File: Server/src/period.ts
interface Period {
    readonly id: number,
    description: string,
    startTime: Date,
    endTime: Date,
    hours: number
}

const updateHours = (period: Period): void => {
    const diff: number = period.endTime.getTime() - period.startTime.getTime();
    period.hours = (diff / 1000 / 60 / 60);
}

const printPeriod = (period: Period): void => {

    console.log("ID: " + period.id)
    console.log("Description: " + period.description)
    console.log("Start Time: " + period.startTime)
    console.log("End Time: " + period.endTime)
    console.log("Duration: " + period.hours + " Hours")
}

export { Period, updateHours, printPeriod }